### PR TITLE
[Feature] add new node about mysql scan node

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
add mysql scan node to scan table from mysql

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/text_converter_column.hpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+#include "vec/common/assert_cast.h"
+namespace doris {
+
+// Note: this function has a codegen'd version.  Changing this function requires
+// corresponding changes to CodegenWriteSlot.
+inline bool TextConverter::write_column(const SlotDescriptor* slot_desc,
+                                        vectorized::MutableColumnPtr* column_ptr, const char* data,
+                                        int len, bool copy_string, bool need_escape,
+                                        MemPool* pool) {
+    //小批量导入只有\N被认为是NULL,没有批量导入的replace_value函数
```
[the remark this answers is not in the document]
Comment (HappenLee) [reply be/src/exec/text_converter_column.hpp:45]: 原来的write_slot的方法呢？

File: be/src/vec/exec/vmysql_scan_node.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+class DescriptorTbl;
+class RowBatch;
+class TextConverter;
+class TupleDescriptor;
+class RuntimeState;
+class MemPool;
+class Status;
+namespace vectorized {
+
+class VMysqlScanNode : public ExecNode {
```
Comment (HappenLee) [inline be/src/vec/exec/vmysql_scan_node.h:37]: 为什么继承了ExecNode，而不是继承mysqlscannode呢？

File: be/src/exec/text_converter_column.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhangstar333) [reply be/src/exec/text_converter_column.hpp:45]: 在text_converter.hpp文件中，没有动这个方法

File: be/src/vec/exec/vmysql_scan_node.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+// specific language governing permissions and limitations
+// under the License.
+#pragma once
+
+#include <memory>
+
+#include "exec/mysql_scan_node.h"
+#include "exec/mysql_scanner.h"
+#include "exec/scan_node.h"
+#include "runtime/descriptors.h"
```
Comment (HappenLee) [inline be/src/vec/exec/vmysql_scan_node.h:24]: 你需要加一些空格让代码更美观

File: be/src/exec/text_converter.hpp [hunk 4]
```diff
@@ -164,6 +164,136 @@ inline bool TextConverter::write_slot(const SlotDescriptor* slot_desc, Tuple* tu
...
+        break;
+    }
+
+    default:
+        DCHECK(false) << "bad slot type: " << slot_desc->type();
+        break;
+    }
+
+    // TODO: add warning for overflow case
+    if (parse_result != StringParser::PARSE_SUCCESS) {
```
Comment (HappenLee) [inline be/src/exec/text_converter.hpp:288]: 你在处理错误时，多加了一个nullable。

**Review by HappenLee — COMMENTED** [review]
有些问题，需要修一下

**Pull request merged**